Helper/Ratings.php

```diff
@@ -128,14 +132,26 @@ private function getRatingsFromProviders(Product $product, Store $store)
                     return null;
                 }
 
+                $activeProduct = $this->registry->registry('current_product');
+
                 try {
+                    if ($activeProduct !== null) {
+                        $this->registry->unregister('current_product');
+                        $this->registry->register('current_product', $product);
+                    } else  {
+                        $this->registry->register('current_product', $product);
+                    }
+
                     /** @noinspection PhpUndefinedMethodInspection */
                     $ratings = $this->ratingsFactory->create()->getRichSnippet();
                 } catch (\Exception $e) {
+                    $this->registry->register('current_product', $activeProduct);
                     $this->logger->exception($e);
                     return null;
                 }
 
+                $this->registry->register('current_product', $activeProduct);
+
                 return [
                     self::AVERAGE_SCORE => $ratings[self::AVERAGE_SCORE],
```

Comment:
I'd add a null check in those array keys since `$this->ratingsFactory->create()->getRichSnippet();` can return an empty string